File: DataGenerator/domain/studentEnrollment.py


class StudentEnrollment:

    def __init__(self, username, fid, year):
        self.__username = username
        self.__fid = fid
        self.__year = year

    @property
    def username(self):
        return self.__username

    @property
    def year(self):
        return self.__year

    @property
    def fid(self):
        return self.__fid

    def __str__(self):
        return "'" + self.__username + "', " + str(self.__fid) + ", " + str(self.__year)

File: AcademicWebApp/src/main/java/com/example/AcademicWebApp/Models/StudentEnrollment.java

package com.example.AcademicWebApp.Models;

import lombok.*;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.IdClass;

@Data
@Entity(name = "studentenrollment")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
@EqualsAndHashCode
@IdClass(StudentEnrollmentId.class)
public class StudentEnrollment {

    @Id
    @Column(name="username")
    private String username;
    @Id
    @Column(name="fid")
    private Integer fid;
    @Id
    @Column(name="year")
    private Integer year;

}
